base/ltoutput.dtx

```diff
@@ -3945,10 +3945,55 @@
      \fi
    \fi
 }
+%    \end{macrocode}
+%  \end{macro}
+%
+%
+%
+%
+%
+%  \begin{macro}{\showfloat}
+%    This command provides some information about the contenta of a
+%    float register.  Float registers are internally named
+%    \cs{bx@}\meta{Uppsercase-letter(s)} and you specify just this
+%    letter or letters as the argument, e.g., \verb=\showfloat{A}=.
+%    (No error recovery no anything if you specify something that
+%    isn't a float.)
+% \changes{v1.4e}{2021/07/31}{Macro added}
+%    \begin{macrocode}
 %</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2021/11/15}%
+%<latexrelease>                 {\showfloat}{Show float register contents}%
+\def\showfloat#1{\begingroup
+    \tracefloats
+    \fl@trace{***Float #1 details:}%
+    \expandafter\fl@showfloat\csname bx@#1\endcsname
+  \endgroup
+}
+\def\fl@showfloat#1{%
+  \fl@traceval{\count#1}%  % this here should be interpreted on day
+  \fl@traceval{\ht#1}%
+  \fl@traceval{\dp#1}%
+  \fl@traceval{\wd#1}%
+  {\tracingonline1\showboxbreadth10\showboxdepth3\showbox#1}%
+}
+%</2ekernel|latexrelease>
+%<latexrelease>\EndIncludeInRelease
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+%<latexrelease>\IncludeInRelease{0000/00/00}%
+%<latexrelease>                 {\showfloat}{Show float register contents}%
+%<latexrelease>
+%<latexrelease>\let\fl@showfloat\@undefined
+%<latexrelease>\EndIncludeInRelease
```

Comment:
Shouldn't this undefine the user-level `\showfloat` instead of the internal macro?  Otherwise people will get an internal error when using `\showfloat` with a rolled-back LaTeX.

Reply:
oops, that was the plan :-)